src/crimson/os/seastore/transaction.h

```diff
@@ -73,7 +73,13 @@ class Transaction {
     if (is_weak()) return;
 
     auto [iter, inserted] = read_set.emplace(this, ref);
-    ceph_assert(inserted);
+    if (!inserted) {
+      ::crimson::get_logger(ceph_subsys_seastore).debug(
+	"{} {} {} already in read_set",
+	__func__,
+	(void*)this,
+	*ref);
+    }
```

Comment:
IIUC the current version of `TransactionManager` and even `SeaStore` doesn't allow concurrent operations within the same transaction, is that correct? @athanatos

Reply:
The segment cleaner would have a different transaction.  Currently, TransactionManager assumes that there won't be concurrent operations on the same transaction, and I'm inclined to leave it that way for now.

Reply:
Oh, sorry, I met this issue in the extent placement manager implementation, in which I parallelized `SegmentCleaner::scan_extents()`. Will remove this commit.